src/compiler.ts

```diff
@@ -8700,27 +8724,114 @@ export class Compiler extends DiagnosticEmitter {
       }
       let fieldInstance = <Field>member;
       let fieldType = fieldInstance.type;
-      exprs[i + 1] = this.module.store( // TODO: handle setters as well
-        fieldType.byteSize,
-        this.module.local_get(tempLocal.index, this.options.nativeSizeType),
-        this.compileExpression(values[i], fieldInstance.type, Constraints.CONV_IMPLICIT),
-        fieldType.toNativeType(),
-        fieldInstance.memoryOffset
+      storeObjectLiteralField(
+        fieldInstance,
+        fieldType,
+        this.compileExpression(values[i], fieldInstance.type, Constraints.CONV_IMPLICIT)
       );
+
+      // This member is no longer omitted, so delete from our omitted fields
+      omittedClassFieldMembers.delete(member.name);
     }
     this.currentType = classReference.type.nonNullableType;
     if (hasErrors) return module.unreachable();
 
+    // Iterate through the remaining omittedClassFieldMembers.
+    if (members) {
+      for(let omittedClassFieldMemberKey of omittedClassFieldMembers.keys()) {
+        let member = members.get(omittedClassFieldMemberKey);
+        
+        if(member) {
+
+          let fieldInstance = <Field>member;
+          let fieldType = fieldInstance.type;
+
+          if (fieldType.classReference) {
+            // TODO: Check if it is a class, with a default value (constructor with no params).
+            // TODO: Check if it can be null, and set to null
+
+            // Otherwise, error
+            this.error(
+              DiagnosticCode.Object_literal_is_missing_class_member_fields_that_must_be_defined,
```

Comment:
This looks like a custom error message. What I typically do is look through https://github.com/microsoft/TypeScript/blob/master/src/compiler/diagnosticMessages.json and check if there is an existing one we can reuse, if we don't have it already. Only if there is none we can reuse, I add a new one.

Reply:
@dcodeIO I looked through the errors (CTRL+F for "literal") and I couldn't find anything that matched. :smile: 

I feel like this error will be needed, even after we add support for nullablity and things, since I expect some classes will have funny stuff with their constructors and things? :thinking: I purposely tried to make it vague enough to be like, Hey, your missing a field that you need to add manually haha!

Reply:
Perhaps, if there is a field on the object that is not on the class type

> `"Property '{0}' does not exist on type '{1}'.": 2339`

and if there is a missing field

> `"Property '{0}' is missing in type '{1}' but required in type '{2}'.": 2741`

with `{1}` being `<object>` and `{2}` being the class type?

Reply:
Dope! Yeah that sounds good, I will use those error messages! :smile: